core/services/vrf/listener_v2.go

```diff
@@ -109,24 +117,63 @@ func (lsn *listenerV2) Start() error {
 			// Do not specify min confirmations, as it varies from request to request.
 		})
 
+		latestHead, unsubscribeHeadBroadcaster := lsn.headBroadcaster.Subscribe(lsn)
+		if latestHead != nil {
+			lsn.setLatestHead(*latestHead)
+		}
+
 		// Log listener gathers request logs
+		lsn.wg.Add(1)
 		go gracefulpanic.WrapRecover(lsn.l, func() {
-			lsn.runLogListener([]func(){unsubscribeLogs}, minConfs)
+			lsn.runLogListener([]func(){unsubscribeLogs}, minConfs, lsn.wg)
 		})
-		var pollPeriod time.Duration
+
+		// Request handler periodically computes a set of logs which can be fulfilled.
+		pollPeriod := 5 * time.Second
 		if lsn.job.VRFSpec.PollPeriod != nil {
 			pollPeriod = *lsn.job.VRFSpec.PollPeriod
-		} else {
-			pollPeriod = 5 * time.Second
 		}
-		// Request handler periodically computes a set of logs which can be fulfilled.
+		lsn.wg.Add(1)
 		go gracefulpanic.WrapRecover(lsn.l, func() {
-			lsn.runRequestHandler(pollPeriod)
+			lsn.runRequestHandler(pollPeriod, lsn.wg)
+		})
+
+		// Head listener updates the latestHeadNumber that we have.
+		lsn.wg.Add(1)
+		go gracefulpanic.WrapRecover(lsn.l, func() {
+			lsn.runHeadListener(unsubscribeHeadBroadcaster, lsn.wg)
 		})
 		return nil
 	})
 }
 
+func (lsn *listenerV2) setLatestHead(head eth.Head) {
+	lsn.latestHeadMu.Lock()
+	defer lsn.latestHeadMu.Unlock()
+	num := uint64(head.Number)
+	if num > lsn.latestHeadNumber {
+		lsn.latestHeadNumber = num
+	}
+}
+
+func (lsn *listenerV2) runHeadListener(unsubscribe func(), wg *sync.WaitGroup) {
```

Comment:
hm I don't think we need this goroutine? We can just pass the head broadcaster unsubscribe into the list of unsubscribes arg in the runLogListener

Reply:
You're right, it's literally just for the unsubscribe. Will remove it.